model/riscv_insts_fext.sail

```diff
@@ -673,22 +659,22 @@ mapping clause encdec =
 /* Execution semantics ================================ */
 
 function clause execute (F_UN_RM_TYPE_S(rs1, rm, rd, FSQRT_S)) = {
-  let rs1_val_S = nan_unbox (F(rs1));
+  let rs1_val_S = X_or_F_S(sys_enable_zfinx(), rs1);
```

Comment:
Can't `X_or_F_S()` call `sys_enable_zfinx()` itself? Unfortunate that we have to keep repeating this all over the place.

Reply:
Yeah that's one of the things I'll be changing

Reply:
(I suspect that's why the register assignment syntax was abandoned, as I could believe Sail not supporting that when you have more than one argument since that doesn't really make sense normally)

Reply:
Hi, was waiting for your feedback, then this went kind of downhill :( Thank you for your valuable help.